mne/io/meas_info.py

```diff
@@ -2478,3 +2478,60 @@ def _write_ch_infos(fid, chs, reset_range, ch_names_mapping):
             for (key, (const, _, write)) in _CH_INFO_MAP.items():
                 write(fid, const, ch[key])
             end_block(fid, FIFF.FIFFB_CH_INFO)
+
+
+def _ensure_infos_match(info1, info2, name, *, on_mismatch='raise'):
+    """Check if infos match.
+
+    Parameters
+    ----------
+    info1, info2 : instance of Info
+        The infos to compare.
+    name : str or int
+        The name of the object appearing in the error message of the comparison
+        fails.
+    on_mismatch : 'raise' | 'warn' | 'ignore'
+        What to do in case of a mismatch between `info1` and `info2`.
+    """
+    _check_option(parameter='on_mismatch', value=on_mismatch,
+                  allowed_values=('raise', 'warn', 'ignore'))
+
+    if not isinstance(name, str):  # passed epochs index
+        name = f'epochs[{name:d}]'
+    info1._check_consistency()
+    info2._check_consistency()
+
+    msg = None
+    if info1['nchan'] != info2['nchan']:
+        msg = f"{name}.info['nchan'] values don't match"
+    elif set(info1['bads']) != set(info2['bads']):
+        msg = f"{name}.info['bads'] don't match"
+    elif info1['sfreq'] != info2['sfreq']:
+        msg = f"{name}.info['sfreq'] don't match"
+    elif set(info1['ch_names']) != set(info2['ch_names']):
+        msg = f"{name}.info['ch_names'] don't match"
+    elif len(info2['projs']) != len(info1['projs']):
+        msg = f"SSP projectors in {name} don't match"
```

Comment:
I don't think we want to do this `msg` for *all* options, just for `info['dev_head_t']`. The others should just directly raise an exception as they did before